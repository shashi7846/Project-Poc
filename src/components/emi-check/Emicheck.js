import React, { useState, useEffect } from "react";
import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faCalendar,
  faPercent,
  faInr,
} from "@fortawesome/free-solid-svg-icons";
import { Link } from "react-router-dom";
import { emiCalculator, emicheckValidation } from "../validators";
import { PuffLoader } from "react-spinners";
import { errorToast } from "../Toast";

const defaultData = {
  loanAmount: "",
  netIncome: "",
  tenure: "",
  price: "",
  interest: "9.5",
};

function Emicheck() {
  const [emicheck, setEmiCheck] = useState(defaultData);
  const [show, setShow] = useState(false);
  const [eligibility, setEligibility] = useState({
    loanAmount: 0,
    emi: 0,
    maxLoanAmount: 0,
    maxEmi: 0,
  });

  const [loading, setLoading] = useState(false);
  useEffect(() => {
    setLoading(true);
    setTimeout(() => {
      setLoading(false);
    }, 2000);
  }, []);

  // let navigate = useNavigate();
  const handleChange = (e) =>
    setEmiCheck({ ...emicheck, [e.target.name]: e.target.value });
  const handleClose = () => setShow(false);

  const handleOnSubmit = (e) => {
    e.preventDefault();
    const errors = emicheckValidation(emicheck);
    if (errors.length) {
      errorToast(errors[0]);
      return;
    }
    const { loanAmount, netIncome, tenure, interest } = emicheck;
    const emi = emiCalculator(loanAmount, interest, tenure);
    const maxLoanAmount = Math.floor(((+netIncome * 0.5) / +emi) * loanAmount);
    const maxEmi = emiCalculator(maxLoanAmount, interest, tenure);
    setEligibility({ loanAmount, emi, maxLoanAmount, maxEmi });
    setShow(true);
  };

  return (
    <>
      {/* {loading ? (
        <PuffLoader
          className="Loader"
          color={'#e60909'}
          loading={loading}
          size={100}
        />
      ) : ( */}
      <div className="container mt-5">
        <div className="d-flex justify-content-center h-100">
          <div className="card bg-dark col-6">
            <div className="card-header ">
              <h3 className="login-name text-light">
                <b>Check Emi</b>
              </h3>
            </div>
            <div className="card-body" style={{ backgroundColor: "#b30000" }}>
              <form onSubmit={handleOnSubmit}>
                <div className="input-group form-group mt-3 mb-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faInr}
                        style={{ fontSize: "1.75em", color: "black" }}
                      />{" "}
                      <label className="text-dark ms-2">
                        <b>PropertyPrice</b>
                      </label>
                    </span>
                  </div>

                  <input
                    type="number"
                    className="form-control ms-2"
                    name="price"
                    placeholder="Enter property price"
                    value={emicheck.price}
                    onChange={handleChange}
                  />
                </div>
                <div className="input-group form-group">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faInr}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>LoanAmount</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    className="form-control ms-2"
                    name="loanAmount"
                    value={emicheck.loanAmount}
                    onChange={handleChange}
                    placeholder="Enter Loan Amount"
                  />
                </div>
                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faInr}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>Income P/M</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    className="form-control ms-2"
                    name="netIncome"
                    value={emicheck.netIncome}
                    onChange={handleChange}
                    placeholder="Net Income Per Month"
                  />
                </div>

                <div className="input-group form-group mt-3">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faCalendar}
                        style={{ fontSize: "1.75em", color: "black" }}
                      ></FontAwesomeIcon>
                      <label className="text-dark ms-2">
                        <b>LoanTenure</b>
                      </label>
                    </span>
                  </div>
                  <input
                    type="number"
                    min={12}
                    max={120}
                    step={12}
                    className="form-control ms-2"
                    name="tenure"
                    value={emicheck.tenure}
                    onChange={handleChange}
                    placeholder=" Select Loan Tenure"
                  />
                </div>
                <div>
                  <label htmlFor="loan-amount" className="form-label"></label>
                  <input
                    type="range"
                    className="form-range mt-2 mb-3"
                    name="tenure"
                    min={12}
                    max={120}
                    step={12}
                    value={emicheck.tenure}
                    onChange={handleChange}
                  ></input>
                </div>

                <div className="input-group form-group">
                  <div className="input-group-prepend">
                    <span
                      className="input-group-text text-white"
                      style={{ minWidth: "150px" }}
                    >
                      <FontAwesomeIcon
                        className="fa-beat-fade"
                        icon={faPercent}
                        style={{ fontSize: "1.75em", color: "black" }}
                      />{" "}
                      <label className="text-dark ms-2">
                        <b>InterestRate</b>
                      </label>
                    </span>
                  </div>

                  <input
                    type="number"
                    aria-label="interest"
                    className="form-control ms-2"
                    value={emicheck.interest}
                    placeholder="Rate Of Interest"
                    disabled
                  />
                </div>
                <div className="form-group mt-2">
                  <Link
                    type="submit"
                    to="/Propertydetails"
                    className="btn float-right  btn-dark mt-2 px-4 "
                  >
                    <b>Back</b>
                  </Link>
                  <Link
                    type="submit"
                    to="/"
                    className="btn float-right  btn-dark mt-2 px-4"
                  >
                    <b>Close</b>
                  </Link>
                  <button
                    type="submit"
                    name="Submit"
                    aria-label="Submit"
                    value="Submit"
                    className="btn float-right  btn-dark mt-2 px-4"
                  >
                    <b>Check Eligibility</b>
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      {/* )} */}

      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Loan Eligibility</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          {/* Emi For Entered Amount will be<b>₹ {eligibility.loanAmount}</b> at EMI
          <b> ₹ {eligibility.emi}</b> */}
          <br /> <h4>You are Eligible for a Maximum loan of</h4> <br />
          <b>₹ {eligibility.maxLoanAmount} </b>
          at EMI <b>₹ {eligibility.maxEmi}</b>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={handleClose}>
            <Link to="/confirmpage" style={{ color: "white" }}>
              {" "}
              Proceed
            </Link>
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
}

export default Emicheck;
