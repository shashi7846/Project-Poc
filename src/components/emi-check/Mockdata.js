export const mockLoginData = {
  PropertyPrice: "10000000",
  LoanAmount: "100000",
  netIncome:"80000",
};
