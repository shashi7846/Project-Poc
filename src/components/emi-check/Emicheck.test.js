import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  renderComponent,
  renderComponentwithRouterPath,
} from "../../utils/test_helpers";
import Emichecktest from "./Emicheck";
import { mockLoginData } from "./Mockdata";
import Emicheck from "./Emicheck";
import Confirmpage from "../confirmpage/Confirmpage";

const user = userEvent.setup();

describe("emicheck page", () => {
  test("intial rendering", () => {
    render(renderComponent(<Emichecktest />));
    expect(screen.getByText("Check Emi")).toBeInTheDocument();
    expect(screen.getByText("LoanAmount")).toBeInTheDocument();
    expect(screen.getByText("PropertyPrice")).toBeInTheDocument();
    expect(
      screen.getByPlaceholderText("Enter property price")
    ).toBeInTheDocument();
  });

  test("intial rendering the links", () => {
    render(renderComponent(<Emichecktest />));
    expect(screen.getByRole("link", { name: "Back" })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: "Close" })).toBeInTheDocument();
  });

  test("should handle input changes in property", async () => {
    render(renderComponent(<Emichecktest />));

    const Property = screen.getByPlaceholderText(/enter property price/i);
    await user.type(Property, mockLoginData.PropertyPrice);
    expect(Property).toHaveValue(+mockLoginData.PropertyPrice);

    const Interest = screen.getByRole("spinbutton", {
      name: /interest/i,
    });
    expect(Interest).toHaveValue(9.5);

    const loanamount = screen.getByPlaceholderText(/enter loan amount/i);
    await user.type(loanamount, mockLoginData.LoanAmount);
    expect(loanamount).toHaveValue(+mockLoginData.LoanAmount);

    const netincome = screen.getByPlaceholderText(/net income per month/i);
    await user.type(netincome, mockLoginData.netIncome);
    expect(netincome).toHaveValue(+mockLoginData.netIncome);
  });

  test("msw testing", async () => {
    renderComponentwithRouterPath(<Emicheck />, "/confirmpage", [
      { path: "/emicheck", element: <Confirmpage /> },
    ]);
    const Property = screen.getByPlaceholderText(/enter property price/i);

    const loanamount = screen.getByPlaceholderText(/enter loan amount/i);

    const netincome = screen.getByPlaceholderText(/net income per month/i);
    const button = screen.getByRole("button", { name: "Submit" });
    await user.type(netincome, mockLoginData.netIncome);
    await user.type(Property, mockLoginData.PropertyPrice);

    await user.type(loanamount, mockLoginData.LoanAmount);

    await user.type(netincome, mockLoginData.netIncome);
    await user.click(button);
    waitFor(() => {
      expect(screen.getByText("Congratulations")).toBeInTheDocument();
    });
  });

  test("Matching the snapshot of the navbar", async () => {
    const { asFragment } = render(renderComponent(<Emichecktest />));
    expect(asFragment()).toMatchSnapshot();
  });
});
