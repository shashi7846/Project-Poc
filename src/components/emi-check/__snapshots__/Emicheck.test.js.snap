// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emicheck page Matching the snapshot of the navbar 1`] = `
<DocumentFragment>
  <div
    class="container mt-5"
  >
    <div
      class="d-flex justify-content-center h-100"
    >
      <div
        class="card bg-dark col-6"
      >
        <div
          class="card-header "
        >
          <h3
            class="login-name text-light"
          >
            <b>
              Check Emi
            </b>
          </h3>
        </div>
        <div
          class="card-body"
          style="background-color: rgb(179, 0, 0);"
        >
          <form>
            <div
              class="input-group form-group mt-3 mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-indian-rupee-sign fa-beat-fade"
                    data-icon="indian-rupee-sign"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 64C0 46.3 14.3 32 32 32H96h16H288c17.7 0 32 14.3 32 32s-14.3 32-32 32H231.8c9.6 14.4 16.7 30.6 20.7 48H288c17.7 0 32 14.3 32 32s-14.3 32-32 32H252.4c-13.2 58.3-61.9 103.2-122.2 110.9L274.6 422c14.4 10.3 17.7 30.3 7.4 44.6s-30.3 17.7-44.6 7.4L13.4 314C2.1 306-2.7 291.5 1.5 278.2S18.1 256 32 256h80c32.8 0 61-19.7 73.3-48H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H185.3C173 115.7 144.8 96 112 96H96 32C14.3 96 0 81.7 0 64z"
                      fill="currentColor"
                    />
                  </svg>
                   
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      PropertyPrice
                    </b>
                  </label>
                </span>
              </div>
              <input
                class="form-control ms-2"
                name="price"
                placeholder="Enter property price"
                type="number"
                value=""
              />
            </div>
            <div
              class="input-group form-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-indian-rupee-sign fa-beat-fade"
                    data-icon="indian-rupee-sign"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 64C0 46.3 14.3 32 32 32H96h16H288c17.7 0 32 14.3 32 32s-14.3 32-32 32H231.8c9.6 14.4 16.7 30.6 20.7 48H288c17.7 0 32 14.3 32 32s-14.3 32-32 32H252.4c-13.2 58.3-61.9 103.2-122.2 110.9L274.6 422c14.4 10.3 17.7 30.3 7.4 44.6s-30.3 17.7-44.6 7.4L13.4 314C2.1 306-2.7 291.5 1.5 278.2S18.1 256 32 256h80c32.8 0 61-19.7 73.3-48H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H185.3C173 115.7 144.8 96 112 96H96 32C14.3 96 0 81.7 0 64z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      LoanAmount
                    </b>
                  </label>
                </span>
              </div>
              <input
                class="form-control ms-2"
                name="loanAmount"
                placeholder="Enter Loan Amount"
                type="number"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-indian-rupee-sign fa-beat-fade"
                    data-icon="indian-rupee-sign"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 64C0 46.3 14.3 32 32 32H96h16H288c17.7 0 32 14.3 32 32s-14.3 32-32 32H231.8c9.6 14.4 16.7 30.6 20.7 48H288c17.7 0 32 14.3 32 32s-14.3 32-32 32H252.4c-13.2 58.3-61.9 103.2-122.2 110.9L274.6 422c14.4 10.3 17.7 30.3 7.4 44.6s-30.3 17.7-44.6 7.4L13.4 314C2.1 306-2.7 291.5 1.5 278.2S18.1 256 32 256h80c32.8 0 61-19.7 73.3-48H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H185.3C173 115.7 144.8 96 112 96H96 32C14.3 96 0 81.7 0 64z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      Income P/M
                    </b>
                  </label>
                </span>
              </div>
              <input
                class="form-control ms-2"
                name="netIncome"
                placeholder="Net Income Per Month"
                type="number"
                value=""
              />
            </div>
            <div
              class="input-group form-group mt-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-calendar fa-beat-fade"
                    data-icon="calendar"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                      fill="currentColor"
                    />
                  </svg>
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      LoanTenure
                    </b>
                  </label>
                </span>
              </div>
              <input
                class="form-control ms-2"
                max="120"
                min="12"
                name="tenure"
                placeholder=" Select Loan Tenure"
                step="12"
                type="number"
                value=""
              />
            </div>
            <div>
              <label
                class="form-label"
                for="loan-amount"
              />
              <input
                class="form-range mt-2 mb-3"
                max="120"
                min="12"
                name="tenure"
                step="12"
                type="range"
                value=""
              />
            </div>
            <div
              class="input-group form-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text text-white"
                  style="min-width: 150px;"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-percent fa-beat-fade"
                    data-icon="percent"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style="font-size: 1.75em; color: black;"
                    viewBox="0 0 384 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M374.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-320 320c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l320-320zM128 128c0-35.3-28.7-64-64-64S0 92.7 0 128s28.7 64 64 64s64-28.7 64-64zM384 384c0-35.3-28.7-64-64-64s-64 28.7-64 64s28.7 64 64 64s64-28.7 64-64z"
                      fill="currentColor"
                    />
                  </svg>
                   
                  <label
                    class="text-dark ms-2"
                  >
                    <b>
                      InterestRate
                    </b>
                  </label>
                </span>
              </div>
              <input
                aria-label="interest"
                class="form-control ms-2"
                disabled=""
                placeholder="Rate Of Interest"
                type="number"
                value="9.5"
              />
            </div>
            <div
              class="form-group mt-2"
            >
              <a
                class="btn float-right  btn-dark mt-2 px-4 "
                href="/Propertydetails"
                type="submit"
              >
                <b>
                  Back
                </b>
              </a>
              <a
                class="btn float-right  btn-dark mt-2 px-4"
                href="/"
                type="submit"
              >
                <b>
                  Close
                </b>
              </a>
              <button
                aria-label="Submit"
                class="btn float-right  btn-dark mt-2 px-4"
                name="Submit"
                type="submit"
                value="Submit"
              >
                <b>
                  Check Eligibility
                </b>
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
